lib/logstash/inputs/syslog.rb

```diff
@@ -194,11 +193,17 @@ def decode(host, output_queue, data)
   end
 
   public
-  def teardown
-    @shutdown_requested.make_true
+  def close
+    super
+    close_udp
+    close_tcp
+  end
+
+  public
+  def stop
+    super
     close_udp
```

Comment:
I wonder if close_udp and close_tcp are necessary here if they are in the close method? does it makes sense? why do we need them in both places?